File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.java
import java.util.HashSet;

class Solution {
    public int solution(int[] elements) {
        int answer = 0;
        int t = 0;
        int l = elements.length;
        HashSet<Integer> hs = new HashSet<>();
        
        for(int i = 0;i<l;i++){
            for(int j = 0;j<l;j++){
                t+=elements[(i+j)%l];
                hs.add(t);
            }
            
            t=0;
        }
        
        answer=hs.size();
        
        return answer;
    }
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
def solution(elements):
    #아니면 미리 (원형 이니깐)>큰 리스트하나를만들어두는것도
    #걍 두번만 반복하면 되나 ?<<<

    #뭔가 중복 연산 ㅠㅣ하는방법

    mySet=set()

    l = len(elements)

    for j in range(l):
        total=0
        for k in range(l):
            total+=elements[(j+k)%l]
            mySet.add(total)   

    return  len(list(mySet))           